python/cudf/cudf/tests/test_cut.py

```diff
@@ -0,0 +1,262 @@
+    "labels", [["bad", "medium", "good"], ["A", "B", "C"], [1, 2, 3], False]
+)
+def test_cut_labels(
+    x, bins, right, include_lowest, ordered, precision, labels
+):
+
+    pcat = pd.cut(
+        x=x,
+        bins=bins,
+        right=right,
+        labels=labels,
+        precision=precision,
+        include_lowest=include_lowest,
+        ordered=ordered,
+    )
+    pindex = pd.CategoricalIndex(pcat) if labels else pcat
+    gindex = cut(
+        x=x,
+        bins=bins,
+        right=right,
+        labels=labels,
+        precision=precision,
+        include_lowest=include_lowest,
+        ordered=ordered,
+    )
+
+    assert_eq(pindex, gindex)
+
+
+@pytest.mark.parametrize("x", [[1, 7, 5, 4, 6, 3]])
+@pytest.mark.parametrize("bins", [3])  # labels must be the same len as bins
+@pytest.mark.parametrize("right", [True, False])
+@pytest.mark.parametrize("include_lowest", [True, False])
+@pytest.mark.parametrize(
+    "ordered", [False]
+)  # labels must be unique if ordered=True
+@pytest.mark.parametrize("precision", [1, 2, 3])
+@pytest.mark.parametrize(
+    "labels", [["bad", "good", "good"], ["B", "A", "B"], [1, 2, 2], False]
+)
+def test_cut_labels_non_unique(
+    x, bins, right, include_lowest, ordered, precision, labels
+):
+
+    pcat = pd.cut(
+        x=x,
+        bins=bins,
+        right=right,
+        labels=labels,
+        precision=precision,
+        include_lowest=include_lowest,
+        ordered=ordered,
+    )
+    pindex = pd.CategoricalIndex(pcat) if labels else pcat
+    gindex = cut(
+        x=x,
+        bins=bins,
+        right=right,
+        labels=labels,
+        precision=precision,
+        include_lowest=include_lowest,
+        ordered=ordered,
+    )
+
+    assert_eq(pindex, gindex)
+
+
+@pytest.mark.parametrize(
+    "x",
+    [
+        [1, 7, 5, 4, 6, 3],
+        [1, 7],
+        np.array([1, 7, 5, 4, 6, 3]),
+        np.array([2, 4, 6, 8, 10]),
+    ],
+)
+@pytest.mark.parametrize(
+    "bins", [1, 2, 3, [1, 2, 3], [0, 2, 4, 6, 10]],
+)
+@pytest.mark.parametrize("right", [True, False])
+@pytest.mark.parametrize("precision", [3])
+def test_cut_right(x, bins, right, precision):
+
+    pcat = pd.cut(x=x, bins=bins, right=right, precision=precision,)
+    pindex = pd.CategoricalIndex(pcat)
+    gindex = cut(x=x, bins=bins, right=right, precision=precision,)
+
+    assert_eq(pindex, gindex)
+
+
+# [1, 2, 2, 3, 3], [0, 2, 4, 6, 10, 10]
+
+
+@pytest.mark.parametrize(
+    "x",
+    [
+        [1, 7, 5, 4, 6, 3],
+        [1, 7],
+        np.array([1, 7, 5, 4, 6, 3]),
+        np.array([2, 4, 6, 8, 10]),
+    ],
+)
+@pytest.mark.parametrize(
+    "bins", [[0, 2, 4, 6, 10, 10], [1, 2, 2, 3, 3]],
+)
+@pytest.mark.parametrize("right", [True, False])
+@pytest.mark.parametrize("include_lowest", [True, False])
+@pytest.mark.parametrize("ordered", [True])
+@pytest.mark.parametrize("precision", [1, 2, 3])
+@pytest.mark.parametrize(
+    "duplicates", ["drop"]
+)  # raise will return a value error
```

Comment:
We actually can test "raise" using `with pytest.raises`

Reply:
Sounds good! I'll test for this!